# HG changeset patch
# User Punarbasu Purkayastha <ppurka@gmail.com>
# Date 1337239403 -28800
# Node ID fc71d0fa6b2cd3599c03a8c8d582697a4b1904e2
# Parent  199c38b75b051010fb474d36e0e49541f069c0c1
set a basic set of legend options so that matplotlib() works

diff --git a/sage/plot/graphics.py b/sage/plot/graphics.py
--- a/sage/plot/graphics.py
+++ b/sage/plot/graphics.py
@@ -273,6 +273,16 @@
         else:
             self._show_legend = bool(show)
 
+        # Add a basic set of legend options to self.__legend_opts so that
+        # self.matplotlib() does not fail.
+        if self.__show_legend:
+            d = dict(back_color=(0.9, 0.9, 0.9), font_family='sans-serif',
+                    font_size='medium', font_style='normal',
+                    font_weight='medium', font_variant='normal')
+            for key in d.iterkeys():
+                if key not in self.__legend_opts:
+                    self.__legend_opts[key] = d[key]
+
     def set_legend_options(self, **kwds):
         r"""
         Set various legend options.
